internal/k8s/crdcontroller.go

```diff
@@ -460,6 +460,16 @@ func validateHTTPRuleObj(key string, httprule *akov1alpha1.HTTPRule) error {
 // validateAviInfraSetting would do validaion checks on the
 // ingested AviInfraSetting objects
 func validateAviInfraSetting(key string, infraSetting *akov1alpha1.AviInfraSetting) error {
+	if ((infraSetting.Spec.Network.EnableRhi != nil && !*infraSetting.Spec.Network.EnableRhi) || infraSetting.Spec.Network.EnableRhi == nil) &&
+		len(infraSetting.Spec.Network.BgpPeerLabels) > 0 {
```

Comment:
You seem to be checking Network.EnableRhi for nil value, can the Network.BgpPeerLabels also be nil? In such a case, will this length check fail?

Reply:
EnableRhi is a pointer to bool which will be nil if not specified. BgpPeerLabels is of type string slice which will be an empty slice if not specified.